src/main/groovy/com/google/gcloud/task/app/AppTask.groovy

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.gcloud.task.app
+import com.google.appengine.AppEnginePlugin
+import com.google.gcloud.task.GCloudTask
+import org.gradle.api.Incubating
+import org.gradle.api.Project
+import org.gradle.api.tasks.Input
+import org.gradle.tooling.BuildException
+
+/**
+ * Generic application task that adds application directory(ies) to the command
+ */
+@Incubating
+class AppTask extends GCloudTask {
+
+    List<Project> modules
+    List<String> descriptors
+
+    @Override
+    protected void configureCommand() {
+        super.configureCommand()
+        (modules as Set).each { Project module ->
```

Comment:
I remember there was a reason for this, maybe because it was a user's input? I'll play around with the code again